### DISCUSSION TITLE
ZCL Work With All Hubs (WWAH) cluster

### DISCUSSION THREAD

**Discussion by MattWestb** [zigpy/zigpy#1177] [ZHA development]
The document number in Zigbee Alliance 17-01067-022 but i cant finding it :-(
But Nordic have commands and attributes described in the info center only click on `WWAH cluster attributes` or `WWAH cluster commands`.
https://infocenter.nordicsemi.com/index.jsp?topic=%2Fsdk_tz_v4.1.0%2Fgroup___z_b___z_c_l___w_w_a_h.html

I dont knowing if ZHA can using this information or if it shall **not** being implanted.

**Comment by dmulcahey** [comment]
I went looking for this doc in the portal and I couldn’t find it either

**Comment by Hedda** [comment]
Looks like additional news about WWAH (Works with All Hubs) was just posted again on [CSA-IOT](https://csa-iot.org) website as if this was a new Zigbee feature:

https://csa-iot.org/newsroom/zigbee-works-with-all-hubs/

https://www.youtube.com/watch?v=Wu7MydmFRFM&ab_channel=ConnectivityStandardsAlliance

Silabs and Texas Instruments also mentioned WWAH cluster will be supported in the Zigbee R23 Specification (also known as Zigbee 2023).

https://e2e.ti.com/support/wireless-connectivity/zigbee-thread-group/zigbee-and-thread/f/zigbee-thread-forum/981579/cc2652p-does-cc2652-support-wwahu-cluster

https://www.silabs.com/documents/login/presentations/tech-talks-futureproofing-your-design-for-project-connected-home-over-ip.pdf

Yet a few years back when it was initially mentioned for the first time it sounded like the work by the "**_All Hubs_**" initiative (a.k.a. "**_All Hubs Initiative_**") workgroup that it would eventually become a part of an upcoming Zigbee 3.1 specification (and not Zigbee 3.0 R23 specs)? 

https://csa-iot.org/newsroom/all-hubs-initiative/

https://mmbnetworks.com/all-hubs-initiative/

https://blog.nordicsemi.com/getconnected/introducing-zigbee-all-hubs

Apparently, Amazon jumped the gun and has since the beginning of December 2020 required that all Zigbee devices must have implemented with [‘Works With All hubs’ (WWAH, a.k.a. WWA and WWAHu)](https://developer.amazon.com/en-US/docs/alexa/smarthome/zigbee-support.html#add-wwahu) cluster in order to be certified by Amazon Echo home hub compatibility and receive the [Works with Alexa (WWA)](https://developer.amazon.com/en-US/docs/alexa/smarthome/zigbee-support.html#wwa-cert) badge. They prove a Works With All Hubs (WWAH) Test Harness for testing via WWA console:

https://developer.amazon.com/en-US/blogs/alexa/device-makers/2020/07/introducing-new-works-with-alexa-requirements-for-zigbee-devices

https://developer.amazon.com/en-US/docs/alexa/smarthome/zigbee-support.html

https://developer.amazon.com/en-US/docs/alexa/smarthome/zigbee-support.html#add-wwahu

https://developer.amazon.com/en-US/docs/alexa/smarthome/zigbee-support.html#test-zigbee

https://developer.amazon.com/en-US/docs/alexa/smarthome/zigbee-support.html#test-wwahu

https://developer.amazon.com/alexa/console/wwa

https://developer.amazon.com/alexa/console/wwa/resources/wwahu

Could the "Works with All Hubs" (WWAH) Zigbee cluster definition be implemented using just those documents from Nordic Semi?

https://infocenter.nordicsemi.com/index.jsp?topic=%2Fsdk_tz_v4.2.0%2Fusing_zigbee__z_c_l.html

https://infocenter.nordicsemi.com/index.jsp?topic=%2Fsdk_tz_v4.2.0%2Fgroup___z_b___z_c_l___w_w_a_h.html

https://infocenter.nordicsemi.com/index.jsp?topic=%2Fsdk_tz_v4.1.0%2Fgroup___z_b___z_c_l___w_w_a_h___a_t_t_r.html

https://infocenter.nordicsemi.com/index.jsp?topic=%2Fsdk_tz_v4.1.0%2Fgroup___z_b___z_c_l___w_w_a_h___c_o_m_m_a_n_d_s.html

Silicon Labs Test Harness module from their Gecko SDK has since 2020 had a SiLabs WWAH Test Harness that should contain a set of python scripts corresponding to each test case specified in the WWAH Test Case Specification.

https://www.silabs.com/documents/login/presentations/tech-talks-futureproofing-your-design-for-project-connected-home-over-ip.pdf

* https://docs.silabs.com/zigbee/7.1/zigbee-af-api/test-harness
  * https://docs.silabs.com/zigbee/7.1/zigbee-af-api/test-harness-cb
    * https://docs.silabs.com/zigbee/7.1/zigbee-af-api/wwah-app-event-retry-manager
    * https://docs.silabs.com/zigbee/7.1/zigbee-af-api/wwah-client-silabs
    * https://docs.silabs.com/zigbee/7.1/zigbee-af-api/wwah-connectivity-manager
    * https://docs.silabs.com/zigbee/7.1/zigbee-af-api/wwah-end-device-extension
    * https://docs.silabs.com/zigbee/7.1/zigbee-af-api/wwah-server-silabs

**Comment by Hedda** [comment]
This announcement from last year by Tuya almost makes it sound as [CSA (Connectivity Standards Alliance)](https://csa-iot.org/) might perhaps eventually release a badge for "WWAH" (Works with All Hubs)  that companies can use on WWAH-certified Zigbee-devices in addition to the Zigbee logo(?):

https://www.tuya.com/news-details/tuya-smart-launches-its-latest-solutions-that-will-support-matter-for-global-customers-Kbw3ecwepppts

"_Tuya, as a long-term ecosystem partner of Amazon, Google, and Apple, helps customers to acquire Matter Badge certifications as well as WWA, WWGA **and WWAH Badge certifications**._"   


Also, Ubisys and Qorvo are now tooting their horn that WWAH has already been implemented by default in their proprietary Zigbee stacks:

https://www.linkedin.com/pulse/adding-amazon-works-all-hubs-your-zigbee-products-arasch-honarbacht/?trk=public_profile_article_view

https://www.ubisys.de/en/support/firmware/changelog-s2r/

"_Fully standard-compliant Works With All Hubs feature support retaining full compatibility with the pre-standard Amazon Works With All Hubs cluster_"

...

"_Enhancements: Core stack changes to support Amazon’s Works With All Hubs cluster_"

Some companies even look to have begun marketing their new Zigbee devices with "WWAH (Works with All Hubs)" instead of "Zigbee 3.0":

https://device.report/manual/4619914

https://device.report/manual/6003507

"_Compatibility: Zigbee, WWAH (Works with All Hubs)_"

PS: Guess if zigpy supported WWAH then all future Zigbee-based "**_Works with Alexa_**" badged devices should be compatible out-of-the-box?

https://developer.amazon.com/en-US/alexa/devices/connected-devices/business-resources/works-with-alexa

![image](https://user-images.githubusercontent.com/6320001/229763920-61953046-06a2-4f15-a376-1affdd37f4a5.png)

PPS: Off-topic but "WWAH" interoperability could maybe also indirectly be of interest to the "**_Works with Home Assistant_**" partner program?

https://www.home-assistant.io/blog/2022/07/12/partner-program/

https://partner.home-assistant.io

![image](https://user-images.githubusercontent.com/6320001/229456974-f7baf348-8cdf-46c1-b3b6-336318ef4794.png)
